core/main_test.ts

```diff
@@ -1131,6 +1165,182 @@ nodes:
           ])
       );
     });
+
+    test(`data preparations resolves defaults before encoding`, () => {
+      const projectDir = createSimpleDataPreparationProjectWithoutDefaults();
+      const dataPreparationYaml = `
+configuration:
+  errorTable:
+    table: error
+  defaults:
+    project: dataprepProject
+    dataset: dataprepDataset
+nodes:
+- id: node1
+  source:
+    table:
+      table: src
+  generated:
+    sourceGenerated:
+      sourceSchema:
+        tableSchema:
+          field:
+          - name: a
+            type: STRING
+            mode: NULLABLE
+    outputSchema:
+      field:
+      - name: a
+        type: INT64
+        mode: NULLABLE
+- id: node2
+  source:
+    nodeId: node1
+  destination:
+    table:
+      table: dest
+  generated:
+    sourceGenerated:
+      sourceSchema:
+        nodeSchema:
+          field:
+          - name: a
+            type: STRING
+            mode: NULLABLE
+    outputSchema:
+      field:
+      - name: a
+        type: INT64
+        mode: NULLABLE
+    destinationGenerated:
+      schema:
+        field:
+        - name: a
+          type: STRING
+          mode: NULLABLE
```

Comment:
Nit: it looks like you only need one node here to cover the test surface, so this YAML file is longer than it needs to be. Suggestion: Remove node2, add a destinationGenerated field to node1.

Reply:
Simplified into a single node.